generated/niscope/session.py

```diff
@@ -2039,6 +2039,113 @@ def get_equalization_filter_coefficients(self):
+
+            relative_to (enums.FetchRelativeTo): Position to start fetching within one record.
+
+            offset (int): Offset in samples to start fetching data within each record. The offset is applied relative to fetch_relative_to. The offset can be positive or negative.
+
+            record_number (int): Zero-based index of the first record to fetch.  Use NUM_RECORDS to set the number of records to fetch.
+
+                Note:
+                One or more of the referenced properties are not in the Python API for this driver.
+
+            num_records (int): Number of records to fetch. Use -1 to fetch all configured records.
+
+            timeout (float): The time to wait for data to be acquired; using 0 for this parameter tells NI-SCOPE to fetch whatever is currently available. Using -1 seconds for this parameter implies infinite timeout.
+
+
+        Returns:
+            wfm_info (list of WaveformInfo): Returns an array of classed with the following timing and scaling information about each waveform:
+
+                -  **relative_initial_x** (float) the time (in seconds) from the trigger to the first sample in the fetched waveform
+                -  **absolute_initial_x** (float) timestamp (in seconds) of the first fetched sample. This timestamp is comparable between records and acquisitions; devices that do not support this parameter use 0 for this output.
+                -  **x_increment** (float) the time between points in the acquired waveform in seconds
+                -  **channel** (str) channel name this waveform was asquire from
+                -  **record** (int) record number of this waveform
+                -  **gain** (float) the gain factor of the given channel; useful for scaling binary data with the following formula:
+
+                    .. math::
+
+                        voltage = binary data * gain factor + offset
+
+                -  **offset** (float) the offset factor of the given channel; useful for scaling binary data with the following formula:
+
+                    .. math::
+
+                        voltage = binary data * gain factor + offset
+
+                - **waveform** (array of float) floating point array of samples. Length will be of the actual samples acquired
+
+        '''
+        import sys
+
+        # Set the fetch attributes
+        with _NoChannel(session=self):
+            self._fetch_relative_to = relative_to
+            self._fetch_offset = offset
+            self._fetch_record_number = record_number
+            self._fetch_num_records = -1 if num_records is None else num_records
+            if num_samples is None:
+                num_samples = self.horz_record_length
+
+        wfm, wfm_info = self._read(num_samples, timeout)
+
+        if sys.version_info.major >= 3:
```

Comment:
From this line to the end of the function, the code is the same as in fetch.
Can it be moved to its own function?

I think something we don't have the ability to do today easily is driver-specific helper functions.

One idea on how to do this is to have a <module>_helpers.py file, and import that automatically form session.py. What do you think?

Reply:
I see that you reused the template for the function, so this is less relevant.
We can revisit the idea if it makes sense in the future.